### ISSUE TITLE
Calendar: When focused on the `Choose Month` and `Choose Year` buttons with a screen-reader, it should also read the selected value [Accessibility]

### ISSUE THREAD

**Issue by Jogonarr-3Pillar** [primefaces/primeng#15849]
### Describe the bug

When focused on the `Choose Month` and `Choose Year` buttons with a screen-reader, it should also read the selected value.

![image](https://github.com/primefaces/primeng/assets/39072072/a2b9dbca-ec80-4c2a-8f87-c02e66b48a89)

When navigating to the buttons, it is narrated that it is a button and what the action of the button does, but it doesn't read the selected value, which is evident for sighted users but not for screen-reader users.

![image](https://github.com/primefaces/primeng/assets/39072072/6b46ca62-3571-4cf1-bc3e-ef47f057c61a)

The reason is the `aria-label` attribute on the buttons does not include the selected value.

### Environment

Requires a screen-reader tool installed (tested witn NVDA v2024.1.0.31547)

![image](https://github.com/primefaces/primeng/assets/39072072/d9de2286-b848-460d-bef3-a9e32fee918b)

Can be replicated with the basic example from the PrimeNg site:
https://primeng.org/calendar#basic

### Reproducer

https://stackblitz.com/edit/github-gw86kd?file=src%2Fapp%2Fapp.component.html

### Angular version

17.3.11

### PrimeNG version

17.18.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.11.1

### Browser(s)

Chrome 125.0.6422.142

### Steps to reproduce the behavior

1. Open a screen-reader (tested wth NVDA v2024.1.0.31547)
2. Tab into the calendar
3. Press down arrow on the keyboard
4. Press tab to go to the Previous Month button
5. Press tab to go to the Choose Month button
6. Notice that it narrates "Choose Month button" but it doesn't narrate the selected value
7. Press tab to go to the Choose Year button
8. Notice that it narrates "Choose Year button" but it doesn't narrate the selected value

### Expected behavior

When focused on the Choose Month button it should read something like: `Choose Month button selected June`.
When focused on the Choose Year button it should read something like: `Choose Year button selected 2024`.

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
